server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -498,8 +498,8 @@ private static void addGeneratedToVerify(Map<ColumnIdent, ColumnConstraint> colu
 
     @SuppressWarnings("unchecked")
     public ParsedDocument index(IndexItem item) throws IOException {
-        assert item.insertValues().length == valueIndexers.size()
-            : "Number of values must match number of targetColumns/valueIndexers";
+        assert item.insertValues().length <= valueIndexers.size()
```

Comment:
in the first attempt, we try to insert, so we might have less values than the indexers size.